packages/components/src/hooks/use-deep-effect.ts

```diff
@@ -3,7 +3,7 @@ import { isDeepEqual } from '@deriv/shared';
 
 // Note: Do not use this effect on huge objects or objects with
 // circular references as performance may suffer.
-export const useDeepEffect = (callback, dependencies) => {
+export const useDeepEffect = (callback: () => void, dependencies: any) => {
     const prev_dependencies = React.useRef(null);
```

Comment:
```suggestion
export const useDeepEffect = (callback: () => void, dependencies: unknown[]) => {
    const prev_dependencies = React.useRef<unknown[] | null>(null);
```

Reply:
dependencies can be an array or an object since isDeepEqual supports both. And `useDeepEffect` is not used anywhere yet. So should it be unknown[]?

Reply:
@likhith-deriv to me, the way dependencies parameter is named suggests that it's a list of dependencies (= array), so it seems logical to pass an array of dependencies to it, and use useDeepEffect as a custom analogue of a real useEffect which accepts an array of dependencies. Yes, sure, isDeepEqual supports any values and it's important to us that it accepts arrays because we want to compare 2 lists (=arrays) of dependencies with its help, so we just say that here we'll pass an array to it. Btw, since useEffect can be called without an array of dependencies at all, it would be even better to make this parameter optional to make it look even nore like a real useEffect 😊:
```suggestion
export const useDeepEffect = (callback: () => void, dependencies?: unknown[]) => {
    const prev_dependencies = React.useRef<unknown[] | null>(null);
```